### PR TITLE
EN6-83 content list

### PR DESCRIPTION
**Pull request by ichalagashvili**
This draft PR contains most features and changes regarding Contents List screen. Its about:
Advanced filtering, filtering via author, status. 
Cloning a content. Joining several contents to categories.
PR containts Edit/Add content small refactorings also.

### CODE REVIEW

File: src/state/contents/actions.js [hunk 1]
```diff
@@ -100,17 +120,80 @@ export const fetchContents = (page = pageDefault,
...
+    if (att === 'categories') {
+      return value.map(
+        (filter, j) => `&filters[${i + j}].attribute=categories&filters[${i + j}].value=${filter.code}`,
+      ).join('');
+    }
+    if (att === 'status' && value === 'published') {
+      return '&status=published';
+    }
+    return `&filters[${i}].attribute=${att}&filters[${i}].value=${value}`;
+  }).join('');
```
Comment (jeffgo10) [inline src/state/contents/actions.js]: you can create all these filtering parameters with this existing tool in entando frontend-libraries (entando-utils). Please check starting at this point https://github.com/entando/frontend-libraries/tree/master/packages/entando-utils#addfilterfilter-string to know about the methods and please utilizing those including future filtering components. I have used these methods in assets/actions, feel free to check them.

File: src/state/contents/actions.js [hunk 2]
```diff
@@ -100,17 +120,80 @@ export const fetchContents = (page = pageDefault,
...
+  const state = getState();
+  const pagination = getPagination(state, 'contents') || getPagination(state);
+  const sortingColumns = getSortingColumns(state);
+  const columnKey = Object.keys(sortingColumns)[0];
+  const sortDirection = sortingColumns[columnKey].direction;
+  const sortParams = `sort=${columnKey}&direction=${sortDirection.toUpperCase()}`;
+  const author = getCurrentAuthorShow(state);
+  const status = getCurrentStatusShow(state);
+  const statusParams = status === 'published' ? '&status=published' : `&filters[0].attribute=status&filters[0].value=${status}`;
+  const authorParams = author === 'all' ? '' : `&filters[0].attribute=author&filters[1].value=${author}`;
```
Comment (jeffgo10) [inline src/state/contents/actions.js]: also this one, use entando-utils filtering methods

File: src/ui/contents/ContentsFilter.js [hunk 3]
```diff
@@ -54,15 +61,13 @@ class ContentsFilter extends Component {
...
+    const { currentQuickFilter, onFilteredSearch } = this.props;
+    const { id, value: currentValue } = currentQuickFilter;
 
     if (keyEvent.key === 'Enter' && currentValue && currentValue.length > 0) {
-      currentQuickFilter.value = '';
-      onSetQuickFilter(currentQuickFilter);
       keyEvent.stopPropagation();
       keyEvent.preventDefault();
-      onFilteredSearch(`?filters[0].attribute=id&filters[0].value=${currentQuickFilter.value}`);
+      onFilteredSearch(`?filters[0].attribute=${id}&filters[0].value=${currentValue}`, null, null, false);
```
Comment (jeffgo10) [inline src/ui/contents/ContentsFilter.js]: use entando-utils filtering params

**Comment by ichalagashvili** [comment]
Latest commit includes fixes in:

- Assets List ( when viewing files, instead of images. Code was accessing undefined img paths when in fact it was unavailable for file)
- Contents List Table column
- Redesigned creation of filtering params

**Pull request merged**